proposals/lip-0051.md

```diff
@@ -769,14 +770,227 @@ if the above send function does not fail:
+                                },
+                                "amount": {
+                                    "dataType": "uint64",
+                                    "fieldNumber": 2
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "supplySubstore": {
+            "type": "array",
+            "fieldNumber": 2,
+            "items": {
+                "type": "object",
+                "required": ["localID", "totalSupply"],
+                "properties": {
+                    "localID": {
+                        "dataType": "uint32",
+                        "fieldNumber": 1
+                    },
+                    "totalSupply": {
+                        "dataType": "uint64",
+                        "fieldNumber": 1
+                    }
+                }
+            }
+        },
+        "escrowSubstore": {
+            "type": "array",
+            "fieldNumber": 3,
+            "items": {
+                "type": "object",
+                "required": ["escrowChainID", "localID", "amount"],
+                "properties": {
+                    "escrowChainID": {
+                        "dataType": "bytes",
+                        "fieldNumber": 1
+                    },
+                    "localID": {
+                        "dataType": "uint32",
+                        "fieldNumber": 2
+                    },
+                    "amount": {
+                        "dataType": "uint64",
+                        "fieldNumber": 3
+                    }
+                }
+            }
+        },
+        "availableLocalIDSubstore": {
+            "dataType": "uint32",
+            "fieldNumber": 4
+        },
+        "terminatedEscrowSubstore": {
+            "type": "array",
+            "fieldNumber": 5,
+            "items": {
+                "type": "uint32"
+            }
+        }
+    }
+}
+```
+
+
+#### Genesis State Initialization
+
+During the genesis state initialization stage, the following steps are executed. If any step fails, the block is discarded and has no further effect.
+
+Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets for the Token module and let `genesisBlockAssetObject` be the deserialization of `genesisBlockAssetBytes` according to the `genesisTokenStoreSchema` schema, given above.
+
+* Initial checks on the properties of `genesisBlockAssetObject`:
+   * Accross all elements of the `userSubstore` array, all `address` values must have length `ADDRESS_LENGTH`. 
+   * Accross all elements of the `userSubstore` array, the pairs `address, tokenID` must be unique (each pair appears at most once, though a given address can appear multiple times with different token IDs). 
+   * For each such `address, tokenID` pair, the values used as `moduleID` in the `lockedBalances` array must be unique.
```

Comment:
Note that in the state store we have an additional requirement:
> In the above object, lockedBalances is always kept ordered by ascending order of moduleID.

Reply:
I added the requirement.